src/App.tsx

```diff
@@ -15,48 +16,119 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((item1, item2) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return item1.localeCompare(item2);
+
+      case SortType.LENGTH:
+        return item1.length - item2.length;
+
+      default:
+        return 0;
+    }
+  });
```

Comment:
I'd prefer to move switch case before `visibleGoods.sort` to iterate like this:
```suggestion
switch (sortType) {
      case SortType.ALPHABET:
        return visibleGood.sort(...);
      case SortType.LENGTH:
        return visibleGoods.sort(...);
      default:
        return visibleGoods;
  });
```
Because your sort type in function doesn’t change, but switch case continue check on each iteration.